BrowserServicesKit/Sources/DDGSync/DDGSync.swift

```diff
@@ -116,7 +115,29 @@ public class DDGSync: DDGSyncing {
         do {
             try await dependencies.createRecoveryKeyTransmitter().send(connectCode)
         } catch {
-            try handleUnauthenticated(error)
+            throw handleUnauthenticatedAndMap(error)
+        }
+    }
+
+    public func createConnectionController(deviceName: String, deviceType: String, delegate: SyncConnectionControllerDelegate) -> SyncConnectionController {
+        SyncConnectionController(deviceName: deviceName, deviceType: deviceType, delegate: delegate, syncService: self, dependencies: dependencies)
+    }
+
+    // Step B
```

Comment:
Remember to remove code comments like this, ideally replacing them with a proper documentation